### PR TITLE
Papd 18 change external references to eit

### PR DESCRIPTION
**Pull request by JDhillonEIT**
Similar to the Client changes, I've bumped the version in use here to signify a breaking change in the bucket reference links and to break from the current versioning.

I've also incorporated the changes we're currently using in [Hostile-S3](https://github.com/GlobalPathogenAnalysisService/hostile-s3/) so this would then supersede that repo.

[Links to this ticket](https://pathogena.atlassian.net/browse/PAPD-18?atlOrigin=eyJpIjoiZDU1ZjFlNjU4NDZlNDBlYWE5YzAwYjBmOTBiY2ZhMDQiLCJwIjoiaiJ9)

--- 

# Additional testing

* Confirmed `hostile fetch` can pull the index files to a new `hostile-eit` application folder. Users can copy across their local files instead of re-downloading if they wish.

* Installed in a local Pathogena Client Conda environment and ran hostile on fastq files

```bash Examples of the above tests.
16:15:19 INFO: EIT Pathogena client version 2.0.0rc1
16:15:19 INFO: Performing FastQ checks and gathering total reads
16:15:19 INFO: Calculating read count in: /Users/jdhillon/code/pathogena/client/tests/data/reads/tuberculosis_1_1.fastq
16:15:19 INFO: Calculating read count in: /Users/jdhillon/code/pathogena/client/tests/data/reads/tuberculosis_1_2.fastq
16:15:19 INFO: 2.0 reads in FASTQ file
16:15:19 INFO: Removing human reads from ILLUMINA FastQ files and storing in /Users/jdhillon/code/pathogena/client
16:15:19 INFO: Hostile version 2.0.0rc1. Mode: paired short read (Bowtie2)
16:15:19 INFO: Fetching standard index human-t2t-hla-argos985-mycob140 (https://objectstorage.uk-london-1.oraclecloud.com/n/lr3yhdniv6gu/b/human-genome-indices/o/human-t2t-hla-argos985-mycob140.tar)
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4.24G/4.24G [12:55<00:00, 5.87MB/s]
16:28:15 INFO: Verifying checksum be0d63ed39d9eeb9657e936c3ded7661c1ef64d8a0de3c1acaf92348c9c8993d…
16:28:25 INFO: Extracting human-t2t-hla-argos985-mycob140.tar…
16:28:28 INFO: Downloaded standard index /Users/jdhillon/Library/Application Support/hostile-eit/human-t2t-hla-argos985-mycob140
16:28:28 INFO: Cleaning…
16:28:29 INFO: Cleaning complete
16:28:29 INFO: Human reads removed from input samples and can be found here: /Users/jdhillon/code/pathogena/client
```

* Tested `pathogena upload` with an Illumina sample with human read removal|decontamination enabled ([sample](https://dev.portal.gpas.world/samples/ab40758e-5161-46c7-b0ec-8c054a29c218))

```bash
16:48:01 INFO: EIT Pathogena client version 2.0.0rc1
16:48:02 INFO: Removing human reads from ILLUMINA FastQ files and storing in /Users/jdhillon/code/pathogena/client
16:48:02 INFO: Hostile version 2.0.0rc1. Mode: paired short read (Bowtie2)
16:48:02 INFO: Found cached standard index human-t2t-hla-argos985-mycob140
16:48:02 INFO: Cleaning…
16:48:29 INFO: Cleaning complete
16:48:29 INFO: Human reads removed from input samples and can be found here: /Users/jdhillon/code/pathogena/client
16:48:31 INFO: Uploading sample1-jay
16:51:43 INFO:   Uploaded ab40758e-5161-46c7-b0ec-8c054a29c218_1.fastq.gz
16:51:43 INFO: Uploading sample1-jay
16:55:09 INFO:   Uploaded ab40758e-5161-46c7-b0ec-8c054a29c218_2.fastq.gz
16:55:09 INFO: Upload complete. Created uzyw9b.mapping.csv (keep this safe)
```

### CODE REVIEW

File: src/hostile/util.py [hunk 1]
```diff
@@ -33,11 +33,11 @@ def choose_default_thread_count(cpu_count: int) -> int:
 CACHE_DIR = (
     Path(os.environ.get("HOSTILE_CACHE_DIR", ""))
     if os.environ.get("HOSTILE_CACHE_DIR")
-    else Path(user_data_dir("hostile", "Bede Constantinides"))
+    else Path(user_data_dir("hostile-eit", "Bede Constantinides"))
 )
 CPU_COUNT = multiprocessing.cpu_count()
 THREADS = choose_default_thread_count(CPU_COUNT)
-BUCKET_URL = "https://objectstorage.uk-london-1.oraclecloud.com/n/lrbvkel2wjot/b/human-genome-bucket/o"
+BUCKET_URL = "https://objectstorage.uk-london-1.oraclecloud.com/n/lr3yhdniv6gu/b/human-genome-indices/o"
```
Comment (JDhillonEIT) [inline src/hostile/util.py:40]: Someone please say if this name for the bucket is correct, chosen as the README mentions the index files in the bucket, can trivially be changed if more appropriate names.

Comment (ngochieu642) [reply src/hostile/util.py:40]: checked with the bucket on gpas-ltd, seems to be correct

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Checkout repository ${{ github.repository }}
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Extract repository name
+        id: repo-name
+        # Strip the owner and `/` from the repo env var
+        run: echo "REPOSITORY_NAME=${GITHUB_REPOSITORY#*/})" >> $GITHUB_ENV
```
Comment (ngochieu642) [inline .github/workflows/release.yml]:
```suggestion
        run: echo "REPOSITORY_NAME=${GITHUB_REPOSITORY#*/}" >> $GITHUB_ENV
```
It seems there is a redundant character here

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Build and push to gpasltd OCR
+        id: build_and_push_gpas_ltd
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          push: true
+          file: Dockerfile
+          tags:
+            "${REGISTRY}/${{ env.REPOSITORY_NAME }}:${{ github.event.release.tag_name }}"
```
Comment (ngochieu642) [inline .github/workflows/release.yml]:
```suggestion
            "${{ env.REGISTRY }}/${{ env.REPOSITORY_NAME }}:${{ github.event.release.tag_name }}"
```

**Review by ngochieu642 — CHANGES_REQUESTED** [review]
Overall looks good to me. Just need to handle some minor syntax in github action script

**Comment by ngochieu642** [comment]
And could you please double check the following things:
- We have a secret `OCI_USERNAME_GPASLTD`
- We have a secret `OCI_TOKEN_GPASLTD`
- The username & token can be used to push to gpasltd
- There is already a `hostile-eit` inside `gpasltd`, we must create this prior to the push, otherwise it will result in some errors

**Comment by JDhillonEIT** [comment]
> And could you please double check the following things:
> 
> * We have a secret `OCI_USERNAME_GPASLTD`
> * We have a secret `OCI_TOKEN_GPASLTD`
> * The username & token can be used to push to gpasltd
> * There is already a `hostile-eit` inside `gpasltd`, we must create this prior to the push, otherwise it will result in some errors

I can confirm these exist and I've tested logging into the registry with these credentials: 
image

I've created `hostile-eit` inside OCIR 

image

**Review by ngochieu642 — APPROVED** [review]

**Pull request merged**